src/main/java/org/opensearch/knn/jni/JNIService.java

```diff
@@ -42,7 +42,7 @@ public static void createIndex(int[] ids, float[][] data, String indexPath, Map<
             return;
         }
 
-        throw new IllegalArgumentException("CreateIndex not supported for provided engine");
+        throw new IllegalArgumentException("CreateIndex not supported for provided engine : " + engineName);
```

Comment:
should we use String.format with param in this case, instead of direct string concatenation